ui/lib/CreateStream.tsx

```diff
@@ -71,60 +162,247 @@ const CreateStream = (props: CreateStreamProps) => {
+          <p>Checking stream token balance...</p>
+        </div>
+      );
+    }
+
+    // 4: Check for enough of the selected token for the stream
+    if (transactionAmount >= tokenBalance) {
+      return (
+        <div>
+          <h3>Not enough tokens to pay for stream</h3>
+          <button
+            onClick={() => {
+              const uniswapURL = `https://app.uniswap.org/#/swap?outputCurrency=${
+                props.tokenAddress
+              }&exactAmount=${
+                transactionAmount - tokenBalance
+              }&exactField=output`;
+              window.open(uniswapURL, "_blank");
+            }}
+          >
+            Go to Uniswap
+          </button>
+        </div>
+      );
+    }
+
+    // 5: If all checks pass, show the slider for stream length
+    return (
+      <div>
+        <h3>Select Stream Length & approve transaction</h3>
+        <div id="CreateStream-lifetime">
+          <Slider
+            min={1}
+            marks={marks}
+            max={maxTimeDays}
+            step={null}
+            value={Math.min(selectedTime / SECS_PER_DAY, maxTimeDays)}
+            defaultValue={1}
+            onChange={(value) =>
+              typeof value === "number" && setSelectedTime(SECS_PER_DAY * value)
+            }
+          />
+        </div>
+        <br />
+        <div id="CreateStream-approve">
+          <button
+            onClick={approveStream}
+            disabled={isSuccess}
+            style={{ backgroundColor: isSuccess ? "grey" : "initial" }}
+          >
+            {`Approve ${transactionAmount} ${tokenData?.symbol}`}
+          </button>
+          {isLoading && <div>Waiting for your confirmation</div>}
+          {isError && <div>You did not confirm the transaction</div>}
+          {txLoading && (
+            <div>
+              Transaction approved; You will be redirected once it has been
+              processed.
+            </div>
+          )}
+          {txError && <div>Transaction process error.</div>}
+        </div>
       </div>
-      <br />
+    );
+  };
+
+  const daysPaid = transactionAmount / props.amountPerSecond;
 
-      {/* Part 3: The user creates their stream, using the given reason/product code */}
+  const Step3 = () => {
+    return (
       <div id="CreateStream-create">
+        <h3>Review params & Open Stream</h3>
+        <div>
+          <p>
+            <strong>Total approval amount:</strong> {transactionAmount}{" "}
+            {tokenData?.symbol}
+          </p>
+          <p>
+            <strong>Days paid:</strong> {daysPaid}
+          </p>
+        </div>
         <button onClick={createStream}>
           {`Open Stream for ${selectedTime / SECS_PER_DAY} day${
             selectedTime !== SECS_PER_DAY ? "s" : ""
           }`}
         </button>
       </div>
-      <br />
+    );
+  };
+
+  return (
+    <div>
+      {currentStep === 0 && <Step1 />}
+      {currentStep === 1 && <Step2 />}
+      {currentStep === 2 && <Step3 />}
```

Comment:
if could move each step outside of the rendering here it would be easier to follow